ghost/webmentions/lib/MentionSendingService.js

```diff
@@ -74,22 +74,24 @@ module.exports = class MentionSendingService {
 
     async send({source, target, endpoint}) {
         logging.info('[Webmention] Sending webmention from ' + source.href + ' to ' + target.href + ' via ' + endpoint.href);
+        
         const response = await this.#externalRequest.post(endpoint.href, {
-            body: {
+            json: {
                 source: source.href,
                 target: target.href,
                 source_is_ghost: true
             },
-            form: true,
             throwHttpErrors: false,
             maxRedirects: 10,
             followRedirect: true,
             methodRewriting: false, // WARNING! this setting has a different meaning in got v12!
```

Comment:
```suggestion
```
Default is true in Got v11, which we want. In got v12 the default is false, but the meaning changed to match that of got v11. Very confusing but the meaning we need here is that a POST should stay a POST after a redirect

Reply:
https://github.com/sindresorhus/got/blob/v11.8.6/readme.md#methodrewriting for ref
I saw some discussion within `got` docs/issues that `post` requests should not be redirected. Not sure if you have any background on that, but that added to the confusion for me here.

Reply:
Correct, in a browser context a POST request generally gets rewritten as a GET. But the webmentions spec (not in a browser context) explicitly states that we should keep the POST method (otherwise it would not work since the body would get lost).

Reply:
Took me a while to figure it out, but there was an error in the commented test:
```
it('Can handle redirect responses', async function () {
            const scope = nock('https://example.org')
                .persist()
                .post('/webmentions-test')
                .reply(302, '', {
                    Location: 'https://example.org/webmentions-test-2'
                });
            const scope2 = nock('https://example.org')
                .persist()
                .post('/webmentions-test-2')
                .reply(201);

            const service = new MentionSendingService({externalRequest});
            const res = await service.send({
                source: new URL('https://example.com'),
                target: new URL('https://example.com'),
                endpoint: new URL('https://example.org/webmentions-test')
            });
            assert(scope.isDone());
            assert(scope2.isDone());
        });
```
The location header was not send by Nock because of this, so that caused the issue. Tests should pass with this change 😊